vars/rapidastUtils.groovy

```diff
@@ -24,6 +24,35 @@ def prepareRapidastStages(String ServiceName, String ApiScanner, String TargetUr
         stage("Collect artifacts") {
             archiveArtifacts allowEmptyArchive: true, artifacts: "results/${ServiceName}/**/zap/*.*, , results.html"
         }
+
+        stage("Send data to Sitreps Grafana") {
+            // There is a a dir that contains a timestamp which would be harder to predict, instead try to find resource.
+            def RAPIDAST_JSON_REPORT = sh returnStdout: true, script: "find . -name zap-report.json"
+            String[] str = RAPIDAST_JSON_REPORT.split("/");
+            def random_results_dir = str[3]
+            def html = "${BUILD_URL}/artifact/results/${ServiceName}/${random_results_dir}/zap/zap-report.html"
+            def json_path = "results/${ServiceName}/${random_results_dir}/zap/zap-report.json"
+            def raw_json = readJSON file: json_path
+            def data = [
+              service: "${ServiceName}",
+              report: raw_json,
+              html_url: html
+            ]
+            def jsonData = groovy.json.JsonOutput.toJson(data)
+            def apiUrl = "https://sitreps-backend-insights-qe-sitreps.apps.ocp-c1.prod.psi.redhat.com/api/v1/rapidast/"
+            def headers = [
+                'Content-type': 'application/json',
+                'Accept': 'text/plain'
+            ]
+            def response = httpRequest(
+                url: apiUrl,
+                httpMode: 'PUT',
+                requestBody: jsonData,
+                headers: headers
+            )
+            echo "Response status: ${response.status}"
+            echo "Response body: ${response.content}"
+        }
```

Comment:
do we want job to fail if it failed to send report to sitreps ?

Reply:
I don't think we need to fail job if sending fails.